test/test_eager_transforms.py

```diff
@@ -720,24 +719,43 @@ def foo(x):
         x = torch.tensor(3.14, device=device)
         functorch.grad(foo)(x)
 
-    @onlyCPU
-    def test_tensor_print(self, device):
-        x = torch.tensor(3.14, device=device)
-        buf = None
-
-        def foo(x):
-            nonlocal buf
-            buf = repr(x)
-            return x
-
-        grad(grad(foo))(x)
-        expected = textwrap.dedent("""\
-                GradTrackingTensor(lvl=3, value=
-                    GradTrackingTensor(lvl=2, value=
-                        tensor(3.1400)
-                    )
-                )""")
-        self.assertEqual(buf, expected)
+    @parametrize("op_list_data", [
+        subtest(([vmap, ], [(64, 3, 32, 32), (4, 2)]), name='vmap'),
+        subtest(([vmap, vmap], [(64, 3, 32, 32), (4, 3, 2)]), name='vmap_vmap'),
+        subtest(([grad, ], [(0, ), [], (4, 2), (64, 3, 32, 32)]), name='grad'),
+        subtest(([grad, grad], [[], ]), name='grad_grad'),
+        subtest(([vmap, grad], [(4, 2)]), name='vmap_grad'),
+    ])
+    def test_tensor_print(self, device, op_list_data):
```

Comment:
@zou3519 how about that ? We can not make data x op product as data depends on the op.

Reply:
This looks great, thanks!